vsintegration/src/FSharp.Editor/CodeLens/AbstractCodeLensDisplayService.fs

```diff
@@ -204,8 +226,12 @@ type CodeLensDisplayService (view : IWpfTextView, buffer : ITextBuffer, layerNam
     
      member self.HandleLayoutChanged (e:TextViewLayoutChangedEventArgs) =
         try
+            // We can cancel existing stuff because the algorithm supports abortion without any data loss
+            self.LayoutChangedCts.Cancel()
+            self.LayoutChangedCts.Dispose()
+            self.LayoutChangedCts <- new CancellationTokenSource()
             let buffer = e.NewSnapshot
-            let recentVisibleLineNumbers = Set [self.RecentLastVsblLineNmbr .. self.RecentLastVsblLineNmbr]
+            let recentVisibleLineNumbers = Set [self.RecentFirstVsblLineNmbr .. self.RecentLastVsblLineNmbr]
```

Comment:
That is the crucial point for the layout problems we have right now.

Reply:
Think you could explain in the PR text how going from Last to First is the crucial bit to fixing a layout issue?